File: app/src/main/java/com/tanxin/base/common/base/BaseActivity.java
package com.tanxin.base.common.base;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;

import com.alibaba.android.arouter.launcher.ARouter;
import com.blankj.utilcode.util.LogUtils;
import com.blankj.utilcode.util.ScreenUtils;
import com.tanxin.base.common.util.Utils;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;

import butterknife.ButterKnife;
import butterknife.Unbinder;
import me.yokeyword.fragmentation.SupportActivity;

/**
 * @author TANXIN
 * @date 2018/6/20/020 
 */
public abstract class BaseActivity extends SupportActivity{
    private Unbinder unbinder;
    protected View contentView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setWindow();
        setBaseView();
        LogUtils.i(getClass().getName());
        unbinder = ButterKnife.bind(this);
        ARouter.getInstance().inject(this);
        EventBus.getDefault().register(this);
        if (ScreenUtils.isPortrait()) {
            ScreenUtils.adaptScreen4VerticalSlide(this, 360);
        } else {
            ScreenUtils.adaptScreen4HorizontalSlide(this, 360);
        }
        initView(savedInstanceState);
    }

    public void setWindow(){}

    protected void setBaseView() {
        setContentView(contentView = LayoutInflater.from(this).inflate(bindLayout(), null));
    }

    /**
     * 获取布局资源
     * @return 布局资源
     */
    protected abstract int bindLayout();


    /**
     * EventBus接收消息
     * @param event 获取事件总线信息
     */
    protected abstract void onEventComing(BaseEvent event);


    /**
     * 初始化界面
     * @param savedInstanceState savedInstanceState
     * */
    protected abstract void initView(Bundle savedInstanceState);



    /**
     * EventBus接收消息
     *
     * @param event 消息接收
     */
    @Subscribe
    public void onEventMainThread(BaseEvent event) {
        if (null != event) {
            onEventComing(event);
        }
    }

    public Context getContext(){
        return this;
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        //解绑ButterKnife
        unbinder.unbind();
        EventBus.getDefault().unregister(this);
    }
}

File: app/src/main/java/com/tanxin/base/newCommon/base/BaseActivity.kt
package com.tanxin.base.newCommon.base

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import com.alibaba.android.arouter.launcher.ARouter
import com.blankj.utilcode.util.LogUtils
import com.blankj.utilcode.util.ScreenUtils
import com.tanxin.base.common.base.BaseEvent
import me.yokeyword.fragmentation.SupportActivity
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe

/**
 * Created by tanxin on 2018/9/28.
 * BaseActivity
 */
abstract class BaseActivity : SupportActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setWindow()
        setBaseView()
        LogUtils.i(javaClass.name)
        ARouter.getInstance().inject(this)
        EventBus.getDefault().register(this)
        if (ScreenUtils.isPortrait()) {
            ScreenUtils.adaptScreen4VerticalSlide(this, 360)
        } else {
            ScreenUtils.adaptScreen4HorizontalSlide(this, 360)
        }
        initView(savedInstanceState)
    }

    open fun setWindow(){
    }

    open fun setBaseView(){
        setContentView(LayoutInflater.from(this).inflate(bindLayout(), null))
    }

    /**
     * 获取布局资源
     * @return 布局资源
     */
    protected abstract fun bindLayout(): Int


    /**
     * EventBus接收消息
     * @param event 获取事件总线信息
     */
    protected abstract fun onEventComing(event: BaseEvent<*>)


    /**
     * 初始化界面
     * @param savedInstanceState savedInstanceState
     */
    protected abstract fun initView(savedInstanceState: Bundle?)

    /**
     * EventBus接收消息
     *
     * @param event 消息接收
     */
    @Subscribe
    fun onEventMainThread(event: BaseEvent<*>?) {
        if (null != event) {
            onEventComing(event)
        }
    }

    fun getContext(): Context {
        return this
    }

    override fun onDestroy() {
        super.onDestroy()
        EventBus.getDefault().unregister(this)
    }
}